DEPRECATION.md

```diff
@@ -19,55 +26,80 @@ underscore (`_`).
+
+- new deprecations to the public API can only occur in minor releases;
 
 - there must always be a way to achieve valid goals that does not issue any
-  warnings;
+  warnings with the most recent two minor releases in a series;
+
+- never assume that an object that is part of the public interface is not in use.
+
+While the no-breaking-changes rule is only formally required *within* a major release series, you should make every effort to avoid breaking changes wherever possible.
+Similarly, while it is permissible where necessary for behavior to change with no single-code path to support both the last minor of one major release and the first minor of a new major release, it is still strongly preferable if you can achieve this.
+
+
+## What is the public interface?
 
-- never assume that a function that isn't explicitly internal isn't in use;
+> [!NOTE]
+> This section should be in sync with [the release schedule documentation of Qiskit](https://docs.quantum.ibm.com/start/install#release-schedule).
+> Please [open an issue against Qiskit](https://github.com/Qiskit/qiskit/issues/new/choose) if there are discrepancies so we can clarify them.
 
-- all deprecations, changes and removals are considered API changes, and can
-  only occur in minor releases not patch releases, per the [stable branch policy](https://github.com/Qiskit/qiskit/blob/main/MAINTAINING.md#stable-branch-policy).
+For the purposes of semantic versioning, the Qiskit public API comprises all *publicly documented* packages, modules, classes, functions, methods, and attributes.
+
+An object is *publicly documented* if and only if it appears in [the hosted API documentation](https://docs.quantum.ibm.com/api/qiskit) for Qiskit.
+The presence of a docstring in the Python source (or a `__doc__` attribute) is not sufficient to make an object publicly documented; this documentation must also be rendered in the public API documentation.
+
+As well as the objects themselves needing to be publicly documented, the only public-API *import locations* for a given object is the location it is documented at in [the public API documentation](https://docs.quantum.ibm.com/api/qiskit), and parent modules or packages that re-export the object (if any).
+For example, while it is possible to import `Measure` from `qiskit.circuit.measure`, this is not a supported part of the public API for two reasons:
+
+1. The module `qiskit.circuit.measure` is not publicly documented, so is not part of the public interface.
+2. The [`Measure` object is documented as being in `qiskit.circuit.library`](https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.Measure), and is re-exported by `qiskit.circuit`, so the public import paths are `from qiskit.circuit.library import Measure` and `from qiskit.circuit import Measure`.
+
+As a rule of thumb, if you are using Qiskit, you should import objects from the highest-level package that exports that object.
```

Comment:
This does raise some questions about things like `QuantumCircuit` and `transpile` which are re-exported from the root `qiskit` import, but are documented in `qiskit.circuit` and `qiskit.compiler` respectively.

Reply:
I'd roughly attempted to address that with bullet point 2.  Are you thinking we need to write more in the public version of the policy / do more in the documentation / something else?

Reply:
TBH, it's more just a discrepancy I hadn't thought of before. More specifically I'm wondering if this points to a deficiency in our documentation more than anything else, should we be documenting the public API as `qiskit.QuantumCircuit` or `qiskit.circuit.QuantumCircuit` and yeah the extension for that for the user facing component of the stability policy which is considered the stable access point. In my head it's both, but I'm wondering how to express that cleanly.